README.md

```diff
@@ -104,6 +104,37 @@ log_request = Promoted::Ruby::Client.prepare_for_logging(request_input)
 log_request.to_json
 ```
 
+## Pass a custom function for compact insertion
+```
+# define a function using proc and it should use one argument.
+# Example
+compact_func = Proc.new do |insertion|
+  insertion.delete(:properties)
```

Comment:
I'm guessing we'll want to copy insertion before deleting.

Reply:
If not set, we should default to not modifying the list.

Reply:
@danbosnichill this is an example function. Users add whatever function they want to add.